client/client.go

```diff
@@ -595,6 +595,12 @@ type TargetSignedStruct struct {
 	Signatures []data.Signature
 }
 
+type NoValidTrustDataError string
```

Comment:
Do we want to name this something more like `NoSuchTarget`?  Because there is already a `ErrRepositoryNotExist` which returns text similar to "... does not have trust data for ...", and the error type name would be clearer I think.